src/client/pmi1.c

```diff
@@ -59,20 +61,37 @@ static int pmi_init = 0;
 
 int PMI_Init(int *spawned)
 {
+    pmix_status_t rc = PMIX_SUCCESS;
     pmix_value_t *val;
-    pmix_status_t rc;
+    pmix_proc_t proc;
+    pmix_info_t info[2];
+    bool  val_optinal = 1;
+    int val_timeout = PMIX_TIMEOUT_VALUE;
 
     if (PMIX_SUCCESS != PMIx_Init(&myproc)) {
         return PMI_ERR_INIT;
     }
 
+    /* getting internal key requires special rank value */
+    memcpy(&proc, &myproc, sizeof(myproc));
+    proc.rank = PMIX_RANK_UNDEF;
+
+    /* set controlling parameters
+     * PMIX_OPTIONAL - expect that these keys should be available on startup
+     * PMIX_TIMEOUT - meet legacy PMI requirement
+     */
+    PMIX_INFO_CONSTRUCT(&info[0]);
+    PMIX_INFO_CONSTRUCT(&info[1]);
+    PMIX_INFO_LOAD(&info[0], PMIX_OPTIONAL, &val_optinal, PMIX_BOOL);
+    PMIX_INFO_LOAD(&info[1], PMIX_TIMEOUT, &val_timeout, PMIX_INT);
```

Comment:
Let's consider a moment what the legacy code's behavior would be in a similar circumstance. When using PMI/PMI2, the data is always collected and returned during the "fence" operation. Thus, any subsequent "get" can be immediately resolved as found or not.

My plan for our backward compatibility library was to provide exactly the same behavior. A call to "fence" from PMI/PMI2 should be a "fence with data return". All subsequent "get" calls should pass an info key directing the server not to seek the data from another location. In other words, we check the client's data store - if not found, we check the local server's data store, and return an error if the data isn't immediately present.

This eliminates the need for an artificial timeout and provides the exact same behavior as PMI/PMI2. I hope that makes sense.